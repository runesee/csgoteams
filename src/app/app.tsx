import Navbar from "../navbar/navbar";
import Mainpage from "../mainpage/mainpage";

export function App() {
  return (
    <>
      <Navbar />
      <Mainpage />
    </>
  );
}
