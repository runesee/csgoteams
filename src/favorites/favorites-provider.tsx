import React, { useEffect, useState } from "react";
import TeamData from "../team/team-data";

interface FavoritesContextValue {
  isFavorite: (team: TeamData) => boolean;
  toggleFavorite: (team: TeamData) => void;
}

interface FavoritesProps {
  children: JSX.Element;
}

export const FavoritesContext = React.createContext<FavoritesContextValue>(
  {} as FavoritesContextValue
);

/*
    Contains state on whether a team is a favorite. This state is stored in local storage, and is kept in sync as long as the state is only changed by toggleFavorite(team); so don't mutate "isFavorites" in local storage directly.
*/
export function FavoritesProvider({ children }: FavoritesProps) {
  const [isFavorite, setIsFavorite] = useState<{ [teamId: number]: boolean }>(
    {}
  );

  //Initialize isFavorite with state from local storage.
  useEffect(() => {
    const maybeIsFavoriteJson: null | string =
      localStorage.getItem("isFavorite");
    if (maybeIsFavoriteJson === null) return;
    setIsFavorite(JSON.parse(maybeIsFavoriteJson));
  }, []);

  return (
    <FavoritesContext.Provider
      value={{
        isFavorite: (team) => isFavorite[team.id] ?? false,
        toggleFavorite: (team) => {
          const toggledIsFavorite = {
            ...isFavorite,
            [team.id]: !(isFavorite[team.id] ?? false)
          };
          //Toggle favorites both in this component's state and in local storage.
          setIsFavorite(toggledIsFavorite);
          localStorage.setItem("isFavorite", JSON.stringify(toggledIsFavorite));
        }
      }}
    >
      {children}
    </FavoritesContext.Provider>
  );
}
