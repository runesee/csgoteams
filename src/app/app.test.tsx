import { describe, expect, test } from "vitest";
import { App } from "./app";
import { render } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { FavoritesProvider } from "../favorites/favorites-provider";

describe("Snapshot test", () => {
  test("Test whether app matches snapshot", () => {
    const app = render(
      <QueryClientProvider client={new QueryClient()}>
        <FavoritesProvider>
          <App />
        </FavoritesProvider>
      </QueryClientProvider>
    );
    expect(app).toMatchSnapshot();
  });
});
