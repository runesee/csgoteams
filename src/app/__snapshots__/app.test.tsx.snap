// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot test > Test whether app matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav"
      >
        <img
          alt="csgo-players"
          class="nav-logo"
          src="logo.png"
        />
      </nav>
      <div
        class="body"
      >
        <div
          class="options"
        >
          <div
            class="filter-and-sort"
          >
            <div
              class="label"
            >
              <p>
                Filter by country: 
              </p>
              <p>
                Sort options: 
              </p>
            </div>
            <div
              class="select"
            >
              <select
                class="dropdown"
                data-testid="selected-country-dropdown"
                id="country"
                name="country"
              >
                <option
                  data-testid="country-select-option"
                  value=""
                >
                  All countries
                </option>
              </select>
              <select
                class="dropdown"
                data-testid="sort-option-dropdown"
                id="sort"
                name="sort"
              >
                <option
                  value="rank-ascending"
                >
                  By rank (ascending)
                </option>
                <option
                  value="rank-descending"
                >
                  By rank (descending)
                </option>
                <option
                  value="name-a-z"
                >
                  By team name (a-z)
                </option>
                <option
                  value="name-z-a"
                >
                  By team name (z-a)
                </option>
              </select>
            </div>
          </div>
          <div
            class="checkboxes"
          >
            <div
              class="label"
            >
              <p>
                Show international teams: 
              </p>
              <p>
                Show only favorite teams: 
              </p>
            </div>
            <div
              class="select"
            >
              <input
                checked=""
                data-testid="international-checkbox"
                id="international"
                name="international"
                type="checkbox"
              />
              <input
                data-testid="favorite-checkbox"
                id="favorite"
                name="favorite"
                type="checkbox"
              />
            </div>
          </div>
        </div>
        <hr
          class="solid"
        />
        <div
          class="teams-container"
          data-testid="teams"
        >
          No teams match your current filters
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav"
    >
      <img
        alt="csgo-players"
        class="nav-logo"
        src="logo.png"
      />
    </nav>
    <div
      class="body"
    >
      <div
        class="options"
      >
        <div
          class="filter-and-sort"
        >
          <div
            class="label"
          >
            <p>
              Filter by country: 
            </p>
            <p>
              Sort options: 
            </p>
          </div>
          <div
            class="select"
          >
            <select
              class="dropdown"
              data-testid="selected-country-dropdown"
              id="country"
              name="country"
            >
              <option
                data-testid="country-select-option"
                value=""
              >
                All countries
              </option>
            </select>
            <select
              class="dropdown"
              data-testid="sort-option-dropdown"
              id="sort"
              name="sort"
            >
              <option
                value="rank-ascending"
              >
                By rank (ascending)
              </option>
              <option
                value="rank-descending"
              >
                By rank (descending)
              </option>
              <option
                value="name-a-z"
              >
                By team name (a-z)
              </option>
              <option
                value="name-z-a"
              >
                By team name (z-a)
              </option>
            </select>
          </div>
        </div>
        <div
          class="checkboxes"
        >
          <div
            class="label"
          >
            <p>
              Show international teams: 
            </p>
            <p>
              Show only favorite teams: 
            </p>
          </div>
          <div
            class="select"
          >
            <input
              checked=""
              data-testid="international-checkbox"
              id="international"
              name="international"
              type="checkbox"
            />
            <input
              data-testid="favorite-checkbox"
              id="favorite"
              name="favorite"
              type="checkbox"
            />
          </div>
        </div>
      </div>
      <hr
        class="solid"
      />
      <div
        class="teams-container"
        data-testid="teams"
      >
        No teams match your current filters
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
